src/odemis/driver/test/xt_client_test.py

```diff
@@ -70,56 +70,75 @@ def setUp(self):
             self.skipTest("No hardware available.")
         if self.microscope.get_vacuum_state() != 'vacuum':
             self.skipTest("Chamber needs to be in vacuum, please pump.")
+        self.xt_type = "xttoolkit" if "xttoolkit" in self.microscope.swVersion.lower() else "xtlib"
 
     def test_acquisition(self):
         """Test acquiring an image."""
         image = self.microscope.get_latest_image(channel_name='electron1')
         self.assertEqual(len(image.shape), 2)
 
-    @unittest.skip("Skip because the microscope stage is not used, an external stage is used.")
     def test_move_stage(self):
         """
         Test that moving the microscope stage to a certain x, y, z position moves it to that position.
         """
-        pos = self.stage.position.value.copy()
+        if self.xt_type == 'xttoolkit':
+            self.skipTest("Skip because the microscope stage is not used, an external stage is used.")
```

Comment:
I don't get the relationship between stage and xttoolkit. Or is it because the current system with xttoolkit has an external stage? 
This might be the case on many systems with a xtlib connection too. Isn't there a way to detect that the stage is missing in xtlib (and xttoolkit)?  In general, it would be handy to fail at initialisation if the user requests a stage component but there is no such hardware. 
Here, if there is no way to check the stage is available, maybe check if the stage is referenced (aka homed)? You could also just leave the unconditional skip, but with a simpler message: "Microscope stage not tested, too dangerous".

Reply:
For now I've gone with "Microscope stage not tested, too dangerous". I did add this point to https://delmic.atlassian.net/browse/BEN-25